File: JANL/SQL/ISQLCommand.vb
Imports System.Data.SqlClient
Imports System.Threading.Tasks

Namespace SQL

	Public Interface ISQLCommand(Of T)
		ReadOnly Property Parameters As SqlParameterCollection
		Property Timeout As Integer

		Function Execute() As T

		Function Execute(Connection As SqlConnection) As T

		Function Execute(Transaction As SqlTransaction) As T

		Function ExecuteAsync() As Task(Of T)

		Function ExecuteAsync(Connection As SqlConnection) As Task(Of T)

		Function ExecuteAsync(Transaction As SqlTransaction) As Task(Of T)

	End Interface

End Namespace

File: JANL/SQL/ISQLCommand.cs
using System.Data.SqlClient;
using System.Threading.Tasks;

#pragma warning disable CS1591 // Отсутствует комментарий XML для открытого видимого типа или члена
namespace JANL.SQL
{
    public interface ISQLCommand<T> : ISQLCommand
    {
        T Execute();

        T Execute(SqlConnection Connection);

        T Execute(string ConnectionString);

        T Execute(SqlTransaction Transaction);

        Task<T> ExecuteAsync();

        Task<T> ExecuteAsync(SqlConnection Connection);

        Task<T> ExecuteAsync(string ConnectionString);

        Task<T> ExecuteAsync(SqlTransaction Transaction);
    }

    public interface ISQLCommand
    {
        SqlParameterCollection Parameters { get; }

        int Timeout { get; set; }
    }
}